lib/astutils.cpp

```diff
@@ -1642,6 +1642,40 @@ struct FwdAnalysis::Result FwdAnalysis::checkRecursive(const Token *expr, const
                 return Result(Result::Type::BAILOUT);
         }
 
+        if (mWhat == What::ValueFlow && Token::simpleMatch(tok, "if (") && Token::simpleMatch(tok->linkAt(1), ") {")) {
+            const Token *bodyStart = tok->linkAt(1)->next();
+            const Token *conditionStart = tok->next();
+            const Token *condTok = conditionStart->astOperand2();
```

Comment:
we have to implement a `getCondition` soon to handle C++ if/while/switch conditions. There's many places with code like this.

Reply:
More importantly, we need a generic forward analysis to unify all of our forward analysis.